src/common/snippets/src/generator.cpp

```
@@ -70,6 +71,9 @@ std::shared_ptr<const TargetMachine> Generator::get_target_machine() const {
 }
 
 RegType Generator::get_op_out_reg_type(const ov::Output<Node>& out) const {
+    auto reg_type = get_specific_op_out_reg_type(out);
+    if (reg_type != RegType::undefined)
+        return reg_type;
```

What will be if backend has specific op with `undefined` reg type (plugin specific `Reshape` or whatever)? Seems like we will throw exception instead of `undefined` returning. Probably we should return boolean status here and make regtype as parameter of function? What do you think?

---

`RegType::undefined` was introduced in this PR to mark that the out reg type is not defined by the backend.
It means that if we'll have plugin-specific `Reshape`, we're gonna need to introduce another type `RegType::ignored` or smth like this. It would med that the reg type is determined by the plugin, but should be ignored by the reg allocation pass.